swapserverrpc/server.proto

```diff
@@ -715,4 +735,128 @@ message ServerWithdrawResponse {
 
     // The nonces that the server used to generate the sweepless sweep sigs.
     repeated bytes server_nonces = 2;
+}
+
+message ServerStaticAddressLoopInRequest {
+    // The client's public key for the htlc output.
+    bytes htlc_client_pub_key = 1;
+
+    // The hashed swap invoice preimage of the swap.
+    bytes swap_hash = 2;
+
+    // The deposit outpoints the client wishes to loop in. They implicitly state
+    // the swap amount.
+    repeated string deposit_outpoints = 3;
+
+    // The swap invoice that the client wants the server to pay.
+    string swap_invoice = 4;
+
+    // An optional last hop the client wants to receive the invoice payment
+    // from.
+    bytes last_hop = 5;
+
+    // The protocol version that the client adheres to.
+    StaticAddressProtocolVersion protocol_version = 6;
+
+    // The user agent string that identifies the software running on the user's
+    // side. This can be changed in the user's client software but it _SHOULD_
+    // conform to the following pattern:
+    //    Agent-Name/semver-version(/additional-info)
+    // Examples:
+    //    loopd/v0.10.0-beta/commit=3b635821
+    //    litd/v0.2.0-alpha/commit=326d754
+    string user_agent = 7;
+}
+
+message ServerStaticAddressLoopInResponse {
+    // The server's public key for the htlc output.
+    bytes htlc_server_pub_key = 1;
+
+    // The cltv expiry height for the htlc. This is the height at which the
+    // htlc will expire and the client is free to claim the funds back.
+    int32 htlc_expiry = 2;
+
+    // The nonces that the server used to generate the partial htlc tx sigs.
+    repeated bytes htlc_server_nonces = 3;
```

Comment:
I propose to group the fields: make a sub-message including two fields: htlc_server_nonces and fee_rate. Then we can use it 3 times here for normal, high and extremely high fee levels.

Reply:
very good suggestion, thanks